### PR TITLE
[MTP] ZMQ Communication Migration and Logprobs Feature Optimization

### PR DESCRIPTION
**Pull request by sunlei1024**
## Motivation

This PR aims to perform an architectural upgrade and capability enhancement for **MTP (Multi-Token Prediction)** in FastDeploy. The core objective is to **introduce a ZMQ-based communication mechanism for MTP scenarios**, reducing dependency on system environment configuration while improving stability and maintainability in multi-process and multi-node deployments.

On top of the communication refactor, this PR also improves **inference observability for MTP** (including `logprobs`, `prompt_logprobs`, and metrics), and optimizes key data processing paths to better support high-throughput inference workloads.

### 1. ZMQ-based Communication Architecture for MTP

* Introduced **ZeroMQ (ZMQ)** as the unified communication mechanism for MTP inference:

  * Reduces dependency on system-level configuration (ports, process topology, deployment constraints)
  * Improves robustness and stability in multi-process and multi-node environments
* Abstracted and unified communication interfaces to enable **shared communication capabilities between MTP and non-MTP scenarios**
* Provides a stable foundation for future MTP extensions (e.g., metrics, asynchronous output)

#### Common Capabilities (Shared by MTP and Non-MTP)

* Unified support for the following inference outputs:

  * `tokens`
  * `logprobs`
  * `prompt_logprobs`
* Aligned pre-processing and post-processing logic across GPU and XPU backends to reduce platform-specific divergence

#### MTP-Specific Capabilities

*  Support for **`draft_token` logprobs**

  * Note: This capability was problematic in the legacy pipeline and therefore disabled by default
  * With the new ZMQ-based architecture, the functionality is re-enabled as an optional capability without affecting existing behavior
* Introduced **metrics collection for MTP scenarios**

  * Enables better observability into MTP inference efficiency, communication status, and performance characteristics

### 3. Data Processing and Performance Optimizations

* Refactored token data processing logic:

  * Replaced complex loop-based logic with `flat` and `cumsum` operations
  * Simplified data structures and reduced intermediate states
* Optimized buffer and memory management:

  * Reduced unnecessary memory copies
  * Improved throughput under high-concurrency inference workloads
* Introduced **asynchronous output via queue-based mechanisms** to further enhance pipeline parallelism

## Impact & Compatibility

This change impacts the following components and execution paths:

* **GPU**

  * Pooling inference pipeline
* **XPU**

  * `prompt_logprobs`
  * `logprobs`
* **EB5**

  * Text-to-image inference pipeline (communication and data processing layers)

There are **no changes to external APIs or user-facing commands**. All existing usage patterns remain fully compatible.

## Usage or Command

No changes.
This PR focuses on internal architectural upgrades and performance optimizations. Existing inference invocation methods remain unchanged.

## Accuracy Tests

The following tests were conducted to validate correctness and stability:

* ✅ MTP basic functionality tests
* ✅ Multi-device compatibility tests (GPU / XPU)
* ✅ ZMQ communication stability and stress tests
* ✅ Accuracy validation for `logprobs` and `prompt_logprobs`

No accuracy regression was observed.

## Checklist

* [x] PR title includes at least one functional tag (e.g., `[Feature]`, `[Optimization]`, `[Speculative Decoding]`, `[XPU]`)
* [x] Code formatted and `pre-commit` checks passed
* [x] Unit tests added or existing coverage verified
* [x] Accuracy results provided
* [ ] If targeting the `release` branch, ensure the PR is first merged into `develop` and cherry-picked with the `[Cherry-Pick]` tag

### CODE REVIEW

File: fastdeploy/spec_decode/mtp.py [hunk 1]
```diff
@@ -1055,24 +1071,37 @@ def _propose_xpu(self, step_use_cudagraph: bool = False, is_dummy_run: bool = Fa
...
+                            accept_token_nums = self.model_inputs["batch_token_num"][:real_bsz]
+                            async_generate_output(
+                                async_output_queue=async_output_queue,
+                                sampled_tokens=sampler_output.sampled_token_ids,
+                                accept_token_nums=accept_token_nums,
+                                logprobs_tensors=sampler_output.logprobs_tensors,
+                                decode_mode=DecodeMode.DRAFT,
+                            )
+                    else:
+                        recover_batch_index_for_sampler_output(sampler_output, self.model_inputs.index_to_batch_id, self.model_inputs.enable_pd_reorder)
```
Comment (Deleter-D) [inline fastdeploy/spec_decode/mtp.py:1087]: 这里是不是重复了

File: fastdeploy/spec_decode/mtp.py [hunk 2]
```diff
@@ -953,29 +957,39 @@ def _propose_cuda(self, step_use_cudagraph: bool = False, is_dummy_run: bool = F
...
+                            accept_token_nums = self.model_inputs["batch_token_num"][:real_bsz]
+                            async_generate_output(
+                                async_output_queue=async_output_queue,
+                                sampled_tokens=sampler_output.sampled_token_ids,
+                                accept_token_nums=accept_token_nums,
+                                logprobs_tensors=sampler_output.logprobs_tensors,
+                                decode_mode=DecodeMode.DRAFT,
+                            )
+                    else:
+                        recover_model_output_map = recover_batch_index_for_output(
```
Comment (Deleter-D) [inline fastdeploy/spec_decode/mtp.py:973]: 走ZMQ是不需要调用这个函数吗？

File: fastdeploy/model_executor/layers/sample/sampler.py [hunk 3]
```diff
@@ -672,9 +672,11 @@ def apply_logits_processor(self, ids: int, future: Optional[Any] = None, prefill
     def compute_logprobs(
         self,
         logits: paddle.Tensor,
-        sampling_metadata: SamplingMetadata,
+        sampling_metadata: Optional[SamplingMetadata] = None,
     ) -> paddle.Tensor:
         """compute logprobs"""
+        if sampling_metadata is None:
+            return F.log_softmax(logits, axis=-1)
```
Comment (Deleter-D) [inline fastdeploy/model_executor/layers/sample/sampler.py:679]: MTPSampler需要同样的修改吗？
